### PR TITLE
[SPARK-6615][MLLIB] Python API for Word2Vec

### PR DESCRIPTION
**Pull request by Lewuathe**
This is the sub-task of SPARK-6254.
Wrap missing method for `Word2Vec` and `Word2VecModel`.

### CODE REVIEW

File: python/pyspark/mllib/tests.py [hunk 1]
```diff
@@ -620,6 +621,42 @@ def test_right_number_of_results(self):
...
+        self.assertEquals(model.minCount, 3)
+
+    def test_word2vec_get_vectors(self):
+        data = [
+            ["I", "have", "a", "pen"],
+            ["I", "like", "soccer", "very", "much"],
+            ["I", "live", "in", "Tokyo"]
+        ]
+        model = Word2Vec().fit(self.sc.parallelize(data))
+        self.assertEquals(len(model.getVectors()), 0)
```
Comment (mengxr) [inline python/pyspark/mllib/tests.py]: Why is the length 0?

File: python/pyspark/mllib/tests.py [hunk 2]
```diff
@@ -620,6 +621,42 @@ def test_right_number_of_results(self):
...
+        model = None
+        try:
+            model = Word2Vec()\
+                .setVectorSize(2)\
+                .setLearningRate(0.01)\
+                .setNumPartitions(2)\
+                .setNumIterations(10)\
+                .setSeed(1024)\
+                .setMinCount(3)
+        except:
```
Comment (mengxr) [inline python/pyspark/mllib/tests.py]: This `try..except..` doesn't do any thing but hide possible exception messages. It is okay to have multiple exceptions and people will check the first one.

File: python/pyspark/mllib/tests.py [hunk 3]
```diff
@@ -620,6 +621,42 @@ def test_right_number_of_results(self):
...
+            ["I", "like", "soccer", "very", "much"],
+            ["I", "live", "in", "Tokyo"]
+        ]
+        model = None
+        try:
+            model = Word2Vec()\
+                .setVectorSize(2)\
+                .setLearningRate(0.01)\
+                .setNumPartitions(2)\
+                .setNumIterations(10)\
```
Comment (mengxr) [inline python/pyspark/mllib/tests.py]: Is 10 too many?

**Comment by mengxr** [comment]
@Lewuathe Could you merge the current master? This PR contains merge conflicts.

**Comment by Lewuathe** [comment]
@mengxr Thank you for reviewing. I merged current master and resolve conflict.

**Comment by mengxr** [comment]
LGTM. Merged into master. (The failed test is irrelevant.)

@andrewor14 @davies I have seen this test failure couple times:

~~~
org.apache.spark.deploy.yarn.YarnClusterSuite.run Python application in yarn-cluster mode
~~~

Is anyone working on it?

**Comment by davies** [comment]
I'm not.

**Comment by andrewor14** [comment]
I filed https://issues.apache.org/jira/browse/SPARK-6701

**Pull request closed without merging**